components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -205,6 +207,14 @@ func (s *Principal) GetTemplateSecrets(ctx context.Context, client sdk.Client, o
 	templateSecrets["ImageRegistryPassword"] = ecrCredentials.Password
 	templateSecrets["ImageRegistryEndpoint"] = ecrCredentials.Endpoint
 
+	assumedRoleCredentials, err := client.GetRoleCredentials(roleArn, time.Hour * 12).Get()
```

Comment:
Not sure why yet but this .Get() breaks when running the tests:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x54 pc=0x91113a]

goroutine 334 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1c0a980, 0x34bd010)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
sync.(*RWMutex).RLock(...)
	/usr/local/go/src/sync/rwmutex.go:48
github.com/aws/aws-sdk-go/aws/credentials.(*Credentials).Get(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/aws/aws-sdk-go@v1.28.12/aws/credentials/credentials.go:227 +0x6a
github.com/alphagov/gsp/components/service-operator/apis/access/v1beta1.(*Principal).GetTemplateSecrets(0xc0009e0000, 0x238e8e0, 0xc000726060, 0x23ad5a0, 0xc000143880, 0xc0003acb10, 0x5, 0x5, 0xc0003acb10)
```

Reply:
probably because you'll need to teach the fake about what your GetRoleCredentials should return